crates/proto/src/rr/domain/name.rs

```diff
@@ -763,6 +766,16 @@ impl Name {
         self.cmp_with_f::<CaseSensitive>(other) == Ordering::Equal
     }
 
+    /// Non-FQDN-aware comparison
+    pub fn lazy_eq(&self, other: &Self) -> bool {
```

Comment:
Where is the `lazy`-name coming from? IMO it would be better to come up with a name that is a little more suggestive of what it does. Maybe `eq_ignore_rooted()` and `eq_ignore_rooted_and_case()`?